subresource-loading.bs

```diff
@@ -31,7 +43,133 @@ specification describes how web browsers load those resources. It is expressed
 as several monkeypatches to the [[HTML]] and [[FETCH]] specification which call
 algorithms defined here.
 
-<p class="note">
-  This specification is under construct. See
-  <a href="https://github.com/WICG/webpackage/issues/708">#708</a>.
-</p>
+Note: This specification is under construct. See <a href="https://github.com/WICG/webpackage/issues/708">#708</a>.
+
+# Structures # {#structures}
+
+A <dfn>fetched web bundle</dfn> is a representation of a web bundle format
+defined in [[draft-yasskin-wpack-bundled-exchanges]].
+
+A <dfn>bundle rule</dfn> is a [=struct=] with the following
+[=struct/items=]:
+
+- <dfn for="bundle rule">source</dfn>, a [=URL=].
+- <dfn for="bundle rule">credentials</dfn>, a [=request/mode=].
+- <dfn for="bundle rule">resources</dfn>, a list of [=URLs=].
+- <dfn for="bundle rule">scopes</dfn>, a list of [=URLs=].
+
+A <dfn>web bundle</dfn> is a [=struct=] with the following [=struct/items=]:
+
+- <dfn for="web bundle">fetched bundle</dfn>, a [=fetched web bundle=] or null.
+- <dfn for="web bundle">rule</dfn>, a [=bundle rule=].
+- <dfn for="web bundle">state</dfn>, an internal state which is "fetching", "fetched", or "failed". Initially "fetching".
+
+A {{Document}} has a <dfn for=document>web bundle list</dfn>, which is a list of [=web bundles=].
+
+# HTML monkeypatches  # {#html-monkeypatches}
+
+Note: TODO. This section uses [=fetch web bundle=].
+
+# Fetch monkeypatches # {#fetch-monkeypatches}
+
+## Monkeypatch HTTP-network-or-cache fetch ## {#monkeypatch-http-network-or-cache-fetch}
+
+In <a spec="fetch">HTTP-network-or-cache fetch</a>, before
+
+> 8.22. Set httpCache to the result of determining the HTTP cache partition, given|httpRequest|.
+
+add the following steps:
+
+22.  Set the |response| to the result of [=fetch from web bundle=], given |httpRequest|.
+
+    1. If |response| is [=network error=], return [=network error=].
+
+    2. If |response| is non-null, skip the steps 8.22-8.24 and goto the step 9.
+
+        Note: That means a subresource from a webbundle never interacts with HttpCache.
+        We plan to support HttpCache as a feature enhancement in the future.
+
+# Algorithms # {#algorithms}
+
+## Fetch web bundle ## {#fetch-web-bundle}
+
+To <dfn id="concept-fetch-web-bundle">fetch web bundle</dfn> given [=web bundle=] |web bundle| and [=fetch params=] |fetch params|:
+
+1. Assert:  |web bundle|'s [=web bundle/state=] is "fetching".
+
+1. Let |request| be |fetch params|'s [=request=].
+
+1. Let |request|'s [=request/url=] is |web bundle|'s [=web bundle/rule=]'s [=bundle rule/source=].
+
+     Note: Source URL is resolved on document's base URL.
+
+1. Let |request|'s [=request/destination=] is "webbundle",
+
+1. Let |request|'s [=request/mode=] is "cors",
+
+1. Let |request|'s [=request/credentials mode=] be |web bundle|'s [=web bundle/rule=]'s [=bundle rule/credentials=] if [=bundle rule/credentials=] is non-null. Otherwise "same-origin".
```

Comment:
Can we wrap at 80 characters (non-strictly), as we do in `loading.bs`?

Reply:
Sure. Done.